### PR TITLE
Mptcp tools

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change
One of the pain point of MPTCP is - because it's not upstream yet - it's kinda hard to configure as standard tools may need to be overriden with their MPTCP fork.
Nix makes it easier.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by teto** [comment]
@layus

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -15685,6 +15687,9 @@ in
   nettools = if stdenv.isLinux then callPackage ../os-specific/linux/net-tools { }
              else unixtools.nettools;
 
+  nettools_mptcp = if stdenv.isLinux then callPackage ../os-specific/linux/net-tools/mptcp.nix { }
+             else unixtools.nettools;
```
Comment (layus) [inline pkgs/top-level/all-packages.nix]: This makes little sense as you would rather see nettols_mptcp fail than installing unixtools.nettools without warnings. This is a good fallback for `nettols` but not for `nettols_mptcp`

**Review by layus — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (teto) [reply pkgs/top-level/all-packages.nix]: I've removed the if.

**Comment by teto** [comment]
I've bumped the versions for mptcp 0.95. The CI errors are strange though ?

File: pkgs/os-specific/linux/iproute/mptcp.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+iproute.overrideAttrs (oa: rec {
+  pname = "iproute_mptcp";
+  version = "0.95";
+
+  src = fetchFromGitHub {
+    owner = "multipath-tcp";
+    repo = "iproute-mptcp";
+    rev = "mptcp_v${version}";
+    sha256 = "070891hfi9z5y4v28gwnyy934ynnanmbi4mh4cx4zhq2azbhbnzc";
```
Comment (layus) [inline pkgs/os-specific/linux/iproute/mptcp.nix]: here ;-), should be 07fihvwlaj0ng8s8sxqhd0a9h1narcnp4ibk88km9cpsd32xv4q3

**Review by layus — CHANGES_REQUESTED** [review]

**Comment by layus** [comment]
With the above change, I could test everything. Seems fine.

**Comment by teto** [comment]
fixed.

**Review by layus — APPROVED** [review]
LGTM :)

**Pull request merged**